### ISSUE TITLE
Does not support nested combineReducers

### ISSUE THREAD

**Issue by maxlapides** [mattkrick/redux-optimistic-ui#19]
Hey there! Thanks so much for building this tool, it's super useful!

We're running into an issue where we have nested `combineReducers` like:

```javascript
combineReducers({
    section1: combineReducers({
        topics: optimistic(topics),
        details
    }),
    section2: combineReducers({
        categories: optimistic(categories),
        geographies
    })
})
```

Unfortunately, it looks like redux-optimistic-ui does not support this kind of thing. The REVERT action will only work for `section1`.  Have you run into this before? Thanks so much!

**Comment by mattkrick** [comment]
Never run into that, happy to fix it if you can find a suitable PR. On the
other hand, why would you ever combineReducers below the root reducer?
You're just introducing unnecessary closures.

**Comment by maxlapides** [comment]
@mattkrick Thanks! Regarding the "why," I got this idea from the Redux docs (http://redux.js.org/docs/api/combineReducers.html):

```
You may call combineReducers at any level of the reducer hierarchy. It doesn't have to happen at the top. In fact you may use it again to split the child reducers that get too complicated into independent grandchildren, and so on.
```

We're using this nested structure to easily separate different routes in our application.

I found a quick fix, and submitted a PR for it. What do you think?

**Comment by mattkrick** [comment]
fixed in v1.0.0 (that's right, 1.0.0. this is now production ready thanks to @maxlapides!)
